### PR TITLE
[LOOM-1287][BpkDismissibleChip][BpkSelectableChip] - Class 2 fixes 

### PR DESCRIPTION
**Pull request by metalix2**
Remember to include the following changes:

- [x] Ensure the PR title includes the name of the component you are changing so it's clear in the release notes for consumers of the changes in the version e.g `[KOA-123][BpkButton] Updating the colour`
- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [ ] Tests
- [ ] Storybook examples created/updated
- [ ] Type declaration (`.d.ts`) files updated
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: packages/bpk-component-chip/src/BpkDismissibleChip.tsx [hunk 1]
```diff
@@ -38,32 +40,58 @@ const getClassName = cssModules(STYLES);
...
-        className={iconClassNames} />}
-      selected
-      type={type}
-      role="button" // Override role="checkbox" because this chip is not selectable.
-      // TODO: className to be removed
-      // eslint-disable-next-line @skyscanner/rules/forbid-component-props
-      className={classNames}
-    />
+    <div className={getClassName(className)}>
+      <button
```
Comment (metalix2) [inline packages/bpk-component-chip/src/BpkDismissibleChip.tsx:61]: best to re-create the BpkSelectableChip as we'd have to pass a classname a wrapping element on an already wrapped Component.
```html
<div classname="custom_classname">
  <div classname="">
    <button classname="bpk_chip">
```

File: packages/bpk-component-chip/src/BpkSelectableChip.tsx [hunk 2]
```diff
@@ -53,36 +53,39 @@ const BpkSelectableChip = ({
     disabled && `bpk-chip--${type}-disabled`,
     !children && 'bpk-chip--icon-only',
     !disabled && selected && `bpk-chip--${type}-selected`,
-    className,
   );
 
   return (
-    <button
-      aria-checked={role === 'button' || role === 'tab' ? undefined : selected}
-      className={classNames}
```
[the remark this answers is not in the document]
Comment (metalix2) [reply packages/bpk-component-chip/src/BpkSelectableChip.tsx]: I think this is an outdated version of the PR?

File: packages/bpk-component-chip/src/BpkSelectableChip.tsx [hunk 3]
```diff
@@ -30,6 +30,7 @@ import STYLES from './BpkSelectableChip.module.scss';
 const getClassName = cssModules(STYLES);
 
 export interface Props extends CommonProps {
+  dismissibleStyle?: boolean;
```
[the remark this answers is not in the document]
Comment (metalix2) [reply packages/bpk-component-chip/src/BpkSelectableChip.tsx]: Prefer dismissible as well.

**Review by mungodewar — APPROVED** [review]
LGTM

**Pull request merged**